test/test_estimators/test_estimators.py

```diff
@@ -496,6 +564,151 @@ def exclude(lst, s):
+
+        # Invalid sort_by column
+        elif (
+            params["sort_by"] not in column_types["all"]
+            and params["sort_by"] != "cost"
+            and params["sort_by"] != ["cost_1", "cost_0"]
+        ):
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # Shouldn't accept an invalid sort order
+        elif params["sort_order"] not in ["ascending", "descending", "auto"]:
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # include is single str but not valid
+        elif (
+            isinstance(params["include"], str)
+            and params["include"] not in column_types["all"]
+        ):
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # Crash if include is list but contains invalid column
+        elif (
+            isinstance(params["include"], list)
+            and len(set(params["include"]) - set(column_types["all"])) != 0
+        ):
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # Can't have just model_id, in both single str and list case
+        elif params["include"] == "model_id" or params["include"] == ["model_id"]:
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # Else all valid combinations should be validated
+        else:
+            leaderboard = model.leaderboard(**params)
+            assert "cost" not in leaderboard.columns
+
+            # top_k should never be less than the rows given back
+            # It can however be larger
+            if isinstance(params["top_k"], int):
+                assert params["top_k"] >= len(leaderboard)
+
+            # Check the right columns are present and in the right order
+            # The model_id is set as the index, not included in pandas columns
+            columns = list(leaderboard.columns)
+
+            def exclude(lst, s):
+                return [x for x in lst if x != s]
+
+            if params["include"] is not None:
+                # Include with only single str should be the only column
+                if isinstance(params["include"], str):
+                    assert params["include"] in columns and len(columns) == 1
+                # Include as a list should have all the columns without model_id
+                else:
+                    assert columns == exclude(params["include"], "model_id")
+            elif params["detailed"]:
+                assert columns == exclude(column_types["detailed"], "model_id")
+            else:
+                assert columns == exclude(column_types["simple"], "model_id")
+
+            # Ensure that if it's ensemble only
+            # Can only check if 'ensemble_weight' is present
+            if params["ensemble_only"] and "ensemble_weight" in columns:
+                assert all(leaderboard["ensemble_weight"] > 0)
+
```

Comment:
[Just a comment]
Looking at this test which I assume it just a modified version of the other one, I wish I hadn't made it so convoluted, it's harder to know what goes wrong. If we have cached estimators between tests, this should be much easier to break up and test specific combos which may be problematic.